main.py

```diff
@@ -229,10 +194,162 @@ def monitor_dns_lookup(test_to_run: TargetHostTest) -> ResultRecord:
+    query_start_time = timestamp_str()
+    # Retrieve the DNS pods
+    ip_addresses = _get_dns_pod_ip_addresses()
+
+    # ICMP ping the DNS server
+    reachable = []
+    print(f'Pinging DNS pods...')
+    success = True
+    for dns_server in ip_addresses:
+        print(f'Pinging {dns_server}')
+        ans = sr1(IP(dst=dns_server) / ICMP(), timeout=1.0, verbose=1)
+        if not ans:
+            print(f'Response: no answer')
+        else:
+            print(f'Response: {ans.summary()}')
+        # retrieve the summary of the answers
+        if ans and ans.haslayer(ICMP) and ans[ICMP].type == 0:
+            print(f"DNS server {dns_server} is reachable")
+            reachable.append(dns_server)
+        else:
+            print(f'DNS server {dns_server} is not reachable')
+        success = False
+    query_end_time = timestamp_str()
+    return ResultRecord(
+        schema_level=SCHEMA_LEVEL,
+        cluster_id=cluster_id,
+        node_name=node_name,
+        process_start_time=process_start_time,
+        test_type=TestType.DNS_LOOKUP.value + test_variant,
+        test_start_time=query_start_time,
+        test_end_time=query_end_time,
+        test_success=success,
+        test_msg=f'ICMP pinged {len(ip_addresses)} DNS pods, {len(reachable)} reachable, {len(ip_addresses)-len(reachable)} unreachable',
+        target_host='',
+        sigints_received=sigints_received,
+        test_extra=json.dumps({
+            'reachable': reachable,
+            'unreachable': [ip for ip in ip_addresses if ip not in reachable],
+        }),
+        node_info=node_info,
+        ci_workload_active=ci_workload_active,
+        ci_workload=ci_workload,
+    )
+
+def prob_dns_pod_liveness_udp(test_to_run: TargetHostTest) -> ResultRecord:
+    global cluster_id, node_name, node_info, ci_workload_active, process_start_time, ci_workload
+    print(f'Checking DNS pod connectivity by UDP...')
+    query_start_time = timestamp_str()
+    # Retrieve the DNS pods
+    ip_addresses = _get_dns_pod_ip_addresses()
+
+    # UDP ping the DNS server
+    print(f'UDP pinging DNS servers...')
+    reachable = []
+    success = True
+    for dns_server in ip_addresses:
+        print(f'UDP pinging {dns_server}')
+        ans = sr1(IP(dst=dns_server) / UDP(dport=0), timeout=1.0, verbose=1)
+        if not ans:
+            print(f'Response: no answer')
+        else:
+            print(f'Response: {ans.summary()}')
+        if ans and ans[ICMP].type == 3 and ans[ICMP].code == 3:
+            print(f'DNS server {dns_server} is reachable')
+            reachable.append(dns_server)
+        else:
+            print(f'DNS server {dns_server} is not reachable')
+            success = False
+    query_end_time = timestamp_str()
+    return ResultRecord(
+        schema_level=SCHEMA_LEVEL,
+        cluster_id=cluster_id,
+        node_name=node_name,
+        process_start_time=process_start_time,
+        test_type=TestType.DNS_LOOKUP.value + test_variant,
```

Comment:
test_types should differ for each of these, no?

Reply:
you are right, I fixed them by changing the test_type